python/src/spark_rapids_ml/classification.py

```diff
@@ -210,3 +284,68 @@ def hasSummary(self) -> bool:
     def numClasses(self) -> int:
         """Number of classes (values which the label can take)."""
         return self._num_classes
+
+    def predict(self, value: Vector) -> float:
+        """
+        Predict label for the given features.
+        """
+        return self.cpu().predict(value)
+
+    def predictLeaf(self, value: Vector) -> float:
+        """
+        Predict the indices of the leaves corresponding to the feature vector.
+        """
+        return self.cpu().predictLeaf(value)
+
+    def predictRaw(self, value: Vector) -> Vector:
+        """
+        Raw prediction for each possible label.
+        """
+        return self.cpu().predictRaw(value)
+
+    def predictProbability(self, value: Vector) -> Vector:
+        """
+        Predict the probability of each class given the features.
+        """
+        return self.cpu().predictProbability(value)
+
+    def evaluate(
```

Comment:
We should consider a gpu version of this where we have all the supporting metrics needed to construct the summary also implemented on gpu.

Reply:
In future work.

Reply:
Good suggestion, Let me file a task for it.